### PR TITLE
Guard IA specific codes with #if defined(__i386__) || defined(__x86_64__)

### PR DESCRIPTION
**Pull request by fenghaitao**

### CODE REVIEW

File: src/common/utils.cpp [hunk 1]
```diff
@@ -19,7 +19,10 @@
 #include <malloc.h>
 #include <windows.h>
 #endif
+
+#ifdef __x86_64__
```
Comment (fenghaitao) [inline src/common/utils.cpp]: mkl-dnn could only be compiled on 64-bit system, so no need to add __i386__.

**Comment by emfomenk** [comment]
Hi @fenghaitao,

Please try promoting the corresponding part to xbyak first, so that we automatically pull-in the feature whenever we update the xbyak version. Once this is done, we will merge Intel MKL-DNN related part.

**Comment by fenghaitao** [comment]
Hi @emfomenk, https://github.com/herumi/xbyak/pull/71 was created to xbyak.

**Comment by fenghaitao** [comment]
Hi @emfomenk, https://github.com/herumi/xbyak/pull/71 was merged into master and herumi modified it a little bit. Will you accept the current PR as it is and resolve the conflict when you rebase xbyak or you want me to rebase xbyak with another PR? The other changes are minimal in mkl-dnn.

**Comment by emfomenk** [comment]
Hi @fenghaitao,

I will take care of that. I am planning to update xbyak inside Intel MKL-DNN, check that everything is oK, and then pull in your PR (with the required adoption to the new version of Intel MKL-DNN).
I would need a few days for that.

Thanks for your work! :tada:

**Comment by emfomenk** [comment]
Thanks @fenghaitao,

Merged!

**Pull request merged**